Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/question/SliderQuestion.kt

```diff
@@ -50,60 +64,130 @@ fun SliderQuestion(
     var sliderPosition by remember {
         mutableStateOf(answer?.answerValue ?: possibleAnswer.defaultValue)
     }
-    Column(modifier = modifier) {
-        Row {
-            Slider(
-                value = sliderPosition,
-                onValueChange = {
-                    sliderPosition = it
-                    onAnswerSelected(it)
-                },
-                valueRange = possibleAnswer.range,
-                steps = possibleAnswer.steps,
-                modifier = Modifier
-                    .padding(horizontal = 16.dp)
-                    .fillMaxWidth()
-            )
+    Column(modifier = modifier.fillMaxSize()) {
+        DiscreteSlider(
+            value = sliderPosition.toInt(),
+            onValueChange = { new ->
+                sliderPosition = new.toFloat()
+                onAnswerSelected(new.toFloat())
+            },
+            startText = stringResource(possibleAnswer.startText),
+            neutralText = stringResource(possibleAnswer.neutralText),
+            endText = stringResource(possibleAnswer.endText)
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DiscreteSlider(
+    value: Int,
+    onValueChange: (Int) -> Unit,
+    startText: String,
+    neutralText: String,
+    endText: String,
```

Comment:
What about passing in a list of options with their labels, and then returning that option as the onValueChanged? If I configured this to have 8 ticks instead of 3, how would this be spaced out?